hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/SplitTableRegionProcedure.java

```diff
@@ -648,9 +656,11 @@ public void createDaughterRegions(final MasterProcedureEnv env) throws IOExcepti
     // there's files to split. It then fires up everything, waits for
     // completion and finally checks for any exception
     //
-    // Note: splitStoreFiles creates daughter region dirs under the parent splits dir
-    // Nothing to unroll here if failure -- re-run createSplitsDir will
-    // clean this up.
+    // Note: From HBASE-26187, splitStoreFiles now creates daughter region dirs straight under the
```

Comment:
The comment is incorrect now? We do not rely on CatalogJanitor to clean up the partial regions. We need to clean in rollback.
I think we should just recreate the whole region directly when we call this method?

Reply:
Yeah, let me remove the CatalogJanitor mention.

>I think we should just recreate the whole region directly when we call this method?

Like, first we check if the given daughter region dir already exists and if so, delete it?

Reply:
I suppose so. Unless we could make sure that every time we will generate the reference files with the same name, then we could just create the reference file with overwrite = true. Otherwise I think we'd better clean the directory and recreate it...

Reply:
Ok, added extra check on HRegionFileSystem.createSplitsDir to delete the daughter regions if those already exist.